### ISSUE TITLE
cert-gen does not generate client certificate

### ISSUE THREAD

**Issue by joshiegy** [coreos/matchbox#690]
`[root@Ansible tls]# ./cert-gen
Creating example CA, server cert/key, and client cert/key...
Generating RSA private key, 4096 bit long modulus
...........++
...........................++
e is 65537 (0x10001)
Generating RSA private key, 2048 bit long modulus
.............................................................................................+++
...............+++
e is 65537 (0x10001)
Using configuration from ./openssl.conf
Error Loading extension section server_cert
139635895134112:error:02001002:system library:fopen:No such file or directory:bss_file.c:175:fopen('./index.txt.attr','rb')
139635895134112:error:2006D080:BIO routines:BIO_new_file:no such file:bss_file.c:182:
139635895134112:error:0E078072:configuration file routines:DEF_LOAD:no such file:conf_def.c:195:
139635895134112:error:0E06D06C:configuration file routines:NCONF_get_string:no value:conf_lib.c:324:group=CA_default name=email_in_dn
139635895134112:error:2207507C:X509 V3 routines:v2i_GENERAL_NAME_ex:missing value:v3_alt.c:531:
139635895134112:error:22098080:X509 V3 routines:X509V3_EXT_nconf:error in extension:v3_conf.c:95:name=subjectAltName, value=10.10.0.120`

[root@Ansible tls]# cat openssl.conf
\# OpenSSL configuration file.
\# Adapted from github.com/dghubble/pegasus

\# default environment variable values
SAN =

[ ca ]
\# `man ca`
default_ca = CA_default

[ CA_default ]
\# Directory and file locations.
dir               = .
certs             = $dir/certs
crl_dir           = $dir/crl
new_certs_dir     = $dir/newcerts
database          = $dir/index.txt
serial            = $dir/serial
\# certificate revocation lists.
crlnumber         = $dir/crlnumber
crl               = $dir/crl/intermediate-ca.crl
crl_extensions    = crl_ext
default_crl_days  = 30
default_md        = sha256

name_opt          = ca_default
cert_opt          = ca_default
default_days      = 375
preserve          = no
policy            = policy_loose

[ policy_loose ]
\# Allow the CA to sign a range of certificates.
countryName             = optional
stateOrProvinceName     = optional
localityName            = optional
organizationName        = optional
organizationalUnitName  = optional
commonName              = supplied
emailAddress            = optional

[ req ]
\# `man req`
default_bits        = 4096
distinguished_name  = req_distinguished_name
string_mask         = utf8only
default_md          = sha256

[ req_distinguished_name ]
countryName                    = Country Name (2 letter code)
stateOrProvinceName            = State or Province Name
localityName                   = Locality Name
0.organizationName             = Organization Name
organizationalUnitName         = Organizational Unit Name
commonName                     = Common Name

\# Certificate extensions (`man x509v3_config`)

[ v3_ca ]
subjectKeyIdentifier = hash
authorityKeyIdentifier = keyid:always,issuer
basicConstraints = critical, CA:true, pathlen:0
keyUsage = critical, digitalSignature, cRLSign, keyCertSign

[ usr_cert ]
basicConstraints = CA:FALSE
nsCertType = client
nsComment = "OpenSSL Generated Client Certificate"
subjectKeyIdentifier = hash
authorityKeyIdentifier = keyid,issuer
keyUsage = critical, nonRepudiation, digitalSignature, keyEncipherment
extendedKeyUsage = clientAuth

[ server_cert ]
basicConstraints = CA:FALSE
nsCertType = server
nsComment = "OpenSSL Generated Server Certificate"
subjectKeyIdentifier = hash
authorityKeyIdentifier = keyid,issuer:always
keyUsage = critical, digitalSignature, keyEncipherment
extendedKeyUsage = serverAuth
subjectAltName = $ENV::SAN"""

What have I done?
Followed the guide.
 120  wget https://github.com/coreos/matchbox/releases/download/v0.7.0/matchbox-v0.7.0-linux-amd64.tar.gz
  121  tar xzvf matchbox-v0.7.0-linux-amd64.tar.gz
  122  cd matchbox-v0.7.0-linux-amd64
  123  cp matchbox /usr/local/bin
  124  ls
  125  useradd -U matchbox
  126  mkdir -p /var/lib/matchbox/assets
  127  chown -R matchbox:matchbox /var/lib/matchbox
  128  cp contrib/systemd/matchbox-local.service /etc/systemd/system/matchbox.service
  129  cd scripts/tls/
  130  export SAN=DNS.1:ansible.fnfd.se,IP1:10.10.0.120
  131  ./cert-gen
  132  history

**Comment by dghubble** [comment]
Try `touch index.txt.attr` like the error indicates. Also, please try to format issues before reporting.

**Comment by dghubble** [comment]
Adding it yourself won't help since the script cleans up that file. Fixed in #695.

Also, you have a typo in your SAN variable.

```
export SAN=DNS.1:ansible.fnfd.se,IP1:10.10.0.120 (bad)
export SAN=DNS.1:ansible.fnfd.se,IP.1:10.10.0.120 (good)
```
